openapi-diff/src/modeler/AutoRest.Swagger/ComparisonMessage.cs

```diff
@@ -15,35 +19,56 @@ public class ComparisonMessage
     {
         public static string DocBaseUrl = "https://github.com/Azure/openapi-diff/tree/master/docs/rules/";
 
-        public ComparisonMessage(MessageTemplate template, FileObjectPath path, Category severity, params object[] formatArguments)
+        public ComparisonMessage(
+            MessageTemplate template,
+            ObjectPath path,
+            IParsedJson old,
+            IParsedJson @new,
+            Category severity,
+            params object[] formatArguments
+        )
         {
             Severity = severity;
             Message = $"{string.Format(CultureInfo.CurrentCulture, template.Message, formatArguments)}";
             Path = path;
+            Old = old;
+            New = @new;
             Id = template.Id;
             Code = template.Code;
             DocUrl = $"{DocBaseUrl}{template.Id}.md";
             Mode = template.Type;
         }
 
+        public IParsedJson Old { get; }
+
+        public IParsedJson New { get; }
+
         public Category Severity { get; }
 
         public string Message { get; }
 
         /// <summary>
         /// The JSON document path to the element being validated.
         /// </summary>
-        public FileObjectPath Path { get; }
+        public ObjectPath Path { get; }
+
+        public string OldJsonRef => Path.JsonPointer(Old);
+
+        public JToken OldJson() => Path.CompletePath(Old.Token).Last().token;
```

Comment:
could you include somewhere a bit of info on JToken and how you're adding location to each element? thanks!